File: javascript/ch1/ex115.js
// Load with load('exXXX.js')

function p(x) {
    return 3 * x - 4 * cube(x);
}

function sine(angle) {
    if (Math.abs(angle) < 0.1) return angle;
    return p(sine(angle / 3.0));
}

function cube(x) {
    return x * x * x;
}

File: erlang/ch1/ex115.erl
%%%-------------------------------------------------------------------
%%% @author Yohan Lee
%%% @copyright (C) 2019
%%% @doc
%%% cd("c:/Users/Yohan/IdeaProjects/sicp/erlang/ch1").
%%% compile with > c("exXXX.erl").
%%% if already compiled, load with > l(exXXX).
%%% run with > exXXX:main().
%%%
%%% @end
%%%-------------------------------------------------------------------
-module(ex115).
-author("Yohan").

%% API
-export([main/0]).

main() ->
  io:fwrite("~w\n", [sine(1)]),
  io:fwrite("~w\n", [sine(2)]),
  io:fwrite("~w\n", [sine(100)]).


cube(X) -> X * X * X.


sine(Angle) when abs(Angle) < 0.1 -> Angle;
sine(Angle) -> p(sine(Angle / 3.0)).


p(X) -> 3 * X - 4 * cube(X).

